app/models/user.rb

```diff
@@ -121,6 +122,46 @@ def owns_project?(project_id)
     projects.exists?(project_id)
   end
 
+  def get_user_preference(key)
+    user_preference = user_preferences.find_by(key: key)
+
+    return user_preference.value unless user_preference.nil?
+
+    return UserPreference::USER_PREFERENCE_METADATA[key][:default]
+  end
+
+  def set_user_preference(key, value)
```

Comment:
I'd prefer this named `save_user_preference` to denote the `save!` call. In general, I think it is better to let the caller decide when to commit changes but not a big deal.